docs/architecture/ADR-003-QGB-ValSet.md

```diff
@@ -0,0 +1,272 @@
+# ADR 002: Overview
+
+## Changelog
+
+- {date}: {changelog}
+
+## Context
+To accommodate the requirements of the [Quantum Gravity Bridge](https://github.com/celestiaorg/quantum-gravity-bridge/blob/master/ethereum/solidity/src/QuantumGravityBridge.sol),
+We will need to add support for `ValSet`s, i.e. Validator Sets, which reflect the current state of the bridge validators.
+
+## Decision
+Add the `ValSet` and `ValSetConfirm` type of messages in order to track the state of the validator set.
+
+## Detailed Design
+Since the QGB is only a one way bridge and is not transferring assets, it doesn't require the portions of the gravity module
+that recreate state from the bridged chain. We only need to keep things relating to signing over the validator set (such as
+`MsgSetOrchestratorAddress` and `MsgValsetConfirm`) and relayer queries (such as `ValsetConfirm` and `GetDelegateKeyByOrchestrator`).
+
+It works by relying on a set of signers to attest to some event on Celestia: the Celestia validator set.
+
+The QGB contract keeps track of the Celestia validator set by updating its view of the validator set with `updateValidatorSet()`.
+More than 2/3 of the voting power of the current view of the validator set must sign off on new relayed events, submitted with
+[`submitDataRootTupleRoot()`](https://github.com/celestiaorg/quantum-gravity-bridge/blob/980b9c68abc34b8d2e4d20ca644b8aa3025a239e/ethereum/solidity/src/QuantumGravityBridge.sol#L328).
+Each event is a batch of `DataRootTuples`, with each tuple representing a single data root (i.e. block header).
+Relayed tuples are in the same order as Celestia block headers.
+For more details, check the data commitment ADR.
+
+Finally, if there are no validator set updates for the unbonding window, the bridge must halt.
+
+### When are validator sets created
+1. If there are no valSet requests, create a new one
+2. If there is at least one validator who started unbonding in current block. (we persist last unbonded block height in `hooks.go`)
+   This will make sure the unbonding validator has to provide an attestation to a new ValSet
+   that excludes him before he completely Unbonds. Otherwise, he will be slashed.
+3. If power change between validators of CurrentValSet and latest valSet request is > 5%
+
+### Message types
+We added the following messages types:
+
+#### Bridge Validator
+The `BridgeValidator` represents a validator's ETH address and its power.
+```protobuf
+message BridgeValidator {
+  uint64 power            = 1;
+  string ethereum_address = 2;
```

Comment:
Why `string` and not `bytes`?

Reply:
Because we will be submitting `0x...` hex encoded addresses